src/requests/filters/models.ts

```diff
@@ -0,0 +1,208 @@
+import { LimitUsageModel } from "../../models/index";
+export interface UpdateFilterInfoBodyParametersModel {}
+
+export interface QueryModel {
+  /**
+   * Partial match for the From: header (case insensitive)
+   */
+  from?: string;
+  /**
+   * Partial match for the To:/Cc: headers (case insensitive)
+   */
+  to?: string;
+  /**
+   * Partial match for the Subject: header (case insensitive)
+   */
+  subject?: string;
+  /**
+   * Partial match for the List-ID: header (case insensitive)
+   */
+  listId?: string;
+  /**
+   * Fulltext search against message text
+   */
+  text?: string;
+  /**
+   * Does a message have to have an attachment or not
+   */
+  ha?: boolean;
+  /**
+   * Message size in bytes. If the value is a positive number then message needs to be larger,
+   * negative then message needs to be smaller than abs(size) value
+   */
+  size?: number;
+}
+
+export interface ActionModel {
+  /**
+   * If true then mark matching messages as Seen
+   */
+  seen?: boolean;
+  /**
+   * If true then mark matching messages as Flagged
+   */
+  flag?: boolean;
+  /**
+   * If true then do not store matching messages
+   */
+  delete?: boolean;
+  /**
+   * If true then store matching messags to Junk Mail folder
+   */
+  spam?: boolean;
+  /**
+   * Mailbox ID to store matching messages to
+   */
+  mailbox?: string;
+  /**
+   * An array of forwarding targets. The value could either be an email address or a relay url to next
+   * MX server ("smtp://mx2.zone.eu:25") or an URL where mail contents are POSTed to
+   */
+  targets?: string[];
+}
+
+export interface CreateNewFilterBodyParametersModel {
```

Comment:
I saw this exact same structure get used in multiple places, you can make this its own interface, put it inside src/models/index.ts and then just extend the interfaces that uses it

Reply:
but only if used outside of filter requests, otherwise it can become its own interface but stay inside this file